libs/parallelism/algorithms/tests/unit/container_algorithms/CMakeLists.txt

```diff
@@ -85,7 +85,8 @@ set(tests
     sort_range
     stable_sort_range
     transform_range
-    transform_range_binary
+    transform_range2
+    #transform_range_binary
```

Comment:
Why is this commented out?

Reply:
This test checks an overload that does not exist in the updated C++20 overloads. There is some discussion need to be made as far as tests are concerned. I added `transform_range2` and I think there are one or two tests that need to be added in order for all the overloads to be checked. Please tell me if you agree in order for me to completeley delete `transform_range_binary`.

Reply:
Sorry for not responding here earlier @gonidelis. I think it'd make sense to leave the test for the overloads that are not part of C++20. This means they'd still be using `hpx::parallel::transform`, `hpx::transform` or `hpx::ranges::transform`. This will lead to deprecation warnings so you'd have to add either `#define HPX_HAVE_DEPRECATION_WARNINGS_V1_6 0` to the test, or `target_compile_definitions(transform_range_binary_test HPX_HAVE_DEPRECATION_WARNINGS_V1_6=0)` to the CMake file. This will of course be deleted once we delete the deprecated overloads but it's unnecessary to delete tests right now that we already have.

Reply:
@msimberg  sorry for responding so late to this, but since we are going to leave just this test as is and since `parallel:transform` invokes `transform_` which returns an `in_out_result` there is some inconsistency here. I could change the result type of `parallel::transform` but it wouldn't be deprecated then. What's the trick?